pyproject.toml

```diff
@@ -0,0 +1,31 @@
+[build-system]
+requires = [
+    "cmake>=3.19",
+    "numpy>=1.21.2",
+    "scikit-build-core >=0.4.3",
+    "ninja; platform_system!='Windows'"
+]
+
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "finufft"
+version = "0.0.1"
+description = "Python bindings for the FINUFFT C++ library"
+readme = "README.md"
+requires-python = ">=3.8"
+
+[tool.scikit-build]
+# Protect the configuration against future changes in scikit-build-core
+minimum-version = "0.4"
+cmake.targets = ["finufft"]
+cmake.args = ['-DFINUFFT_BUILD_PYTHON=ON', '-DCMAKE_BUILD_TYPE=Release'] # not sure if this is necessary
+wheel.packages = ["python/finufft/finufft"]
```

Comment:
How does this know to pick up the `libfinufft.so` file?

Reply:
From:
```
    install(TARGETS finufft LIBRARY DESTINATION finufft)
```

Reply:
But doesn't that line tell cmake to install libfinufft.so in the `python/finufft` directory (not the `python/finufft/finufft` directory)? I guess it must be doing something else.

Reply:
So from what I understand, this is what skbuild does, essentially:
1. Runs `cmake` and `cmake --build`
2. Runs `cmake --install` to install the targets into some temporary directory (on my system, something like `/tmp/tmpn4jdyywf/wheel/platlib`).
3. Copies the package in `wheels.packages` into that directory
4. Zips that directory up into a `.whl` file

So in our case, step 2 will install `finufft/libfinufft.so` into that temp directory, while step 3 will copy `python/finufft/finufft` there, which results in all the files living in the right places (so `libfinufft.so` will be in the same directory as `_finufft.py` and friends). This means that step 4 will put everything together into the desired wheel.